### ISSUE TITLE
app only access database when connected to WiFi

### ISSUE THREAD

**Issue by GoogleCodeExporter** [nelenkov/wwwjdic#41]
```
Dear developer,

On my HTC DesireZ the app can only access the online database (e.g. SOD but 
also the dictionary itself, when connected to WiFi. When connected to 3G/GPRS 
it gives "0 results" on any query.

Also the Kanji of the day will update only on WiFi.

With kindest regards,
Ronald.
Newsgroup@hilhorst.eu


```

Original issue reported on code.google.com by `ronald.h...@j-link.nl` on 20 Apr 2011 at 5:05

**Comment by GoogleCodeExporter** [comment]
```
Hi, 

Thanks for your report. The app doesn't really know anything about WiFi or 3G, 
it just uses the Internet connection the phone provides. Does browsing, etc. 
work for you on 3G?

One thing you could try is change the mirror site to see if that makes a 
difference.
Go to Settings, uncheck 'Mirror auto select' and change the mirror site below. 
Try different ones to see what happens. 
```

Original comment by `nikolay....@gmail.com` on 20 Apr 2011 at 5:19

**Comment by GoogleCodeExporter** [comment]
```
Same problem here. The home screen widget doesn't work either whe.n the 
connection isn't through wifi. Changing the mirror doesn't seem to have any 
effect.
```

Original comment by `m.debeni...@gmail.com` on 29 Jun 2011 at 6:23

**Comment by GoogleCodeExporter** [comment]
```
I forgot: I'm running the app on a stock Samsung Galaxy S2.
```

Original comment by `m.debeni...@gmail.com` on 29 Jun 2011 at 6:25

**Comment by GoogleCodeExporter** [comment]
```
Is this a problem only with the widget, or does search via the app not work as 
well?
```

Original comment by `nikolay....@gmail.com` on 30 Jun 2011 at 3:35

**Comment by GoogleCodeExporter** [comment]
```
It would help if you could send me a log file from the time the error? happens.
Here's how:

1. Get the Log Collector app from market 
(https://market.android.com/details?id=com.xtralogic.android.logcollector)
2. Install the widget, wait a couple of minutes to give it chance to try to 
update.
3. Start Log Collector, send the log to yourself and then paste it here (or 
send it directly to me if you want).

Thanks.

```

Original comment by `nikolay....@gmail.com` on 30 Jun 2011 at 3:45

**Comment by GoogleCodeExporter** [comment]
```
Hi, sorry for the delay. I've downloaded the log collector app but it 
apparently truncates the log when copying it to the mail app and all I see it's 
a bunch of touch events dispatched, then an abrupt end to the log. No reference 
whatsoever to network access or your app. What I did is: launch  wwwjdic, tap 
on any kanji, perform any search or do anything that requires net access. I get 
"0 results". I then launch the log app, send as email, read the email, nothing. 
I've repeated this a few times until I got something other than tap events, and 
though it still seems pretty unhelpful to me, at least now there are some 
network events. I hope it helps.

 Log Collector version: 1.1.0
 Device model: GT-I9100
 Firmware version: 2.3.3
 Kernel version: 2.6.35.7-I9100BUKE2-CL188370
root@DELL102 )
#2 SMP PREEMPT Fri May 6 16:37:20 KST 2011
 Build number: GINGERBREAD.BUKE2

ispatcher( 2689): Delivering touch to current input target: action: 1, channel 
'40ae0c28 InputMethod (server)'
07-11 11:21:37.545 I/InputDispatcher( 2689): Delivering touch to current input 
target: action: 1, channel 'PointerDevice-11619 (server)'
07-11 11:21:37.574 E/webtextview(11619): ....Doing a 
super.requestRectangleOnScreen....
07-11 11:21:38.070 D/FastDormancy( 2811):  before ======= ENTER DORMANCY =======
07-11 11:21:38.070 D/FastDormancy( 2811): [FD] ON default: true
07-11 11:21:38.075 D/FastDormancy( 2811): Before mDormFlag: false in 
getCapaDormancy()
07-11 11:21:38.075 D/FastDormancy( 2811): After mDormFlag: false in 
getCapaDormancy()
07-11 11:21:38.675 W/PowerManagerService( 2689): Timer 0x3->0x3|0x0
07-11 11:21:39.074 D/FastDormancy( 2811):  before ======= ENTER DORMANCY =======
07-11 11:21:39.074 D/FastDormancy( 2811): [FD] ON default: true
07-11 11:21:39.079 D/FastDormancy( 2811): Before mDormFlag: false in 
getCapaDormancy()
07-11 11:21:39.089 D/FastDormancy( 2811): After mDormFlag: false in 
getCapaDormancy()
07-11 11:21:40.095 D/FastDormancy( 2811):  before ======= ENTER DORMANCY =======
07-11 11:21:40.095 D/FastDormancy( 2811): [FD] ON default: true
07-11 11:21:40.095 D/FastDormancy( 2811): Before mDormFlag: false in 
getCapaDormancy()
07-11 11:21:40.095 D/FastDormancy( 2811): After mDormFlag: false in 
getCapaDormancy()
07-11 11:21:41.100 D/FastDormancy( 2811):  before ======= ENTER DORMANCY =======
07-11 11:21:41.100 D/FastDormancy( 2811): [FD] ON default: true
07-11 11:21:41.100 D/FastDormancy( 2811): Before mDormFlag: false in 
getCapaDormancy()
07-11 11:21:41.100 D/FastDormancy( 2811): After mDormFlag: false in 
getCapaDormancy()
07-11 11:21:41.325 I/TelephonyRegistry( 2689): notifyDataConnection: state=2 
activeState=2 isDataConnectivityPossible=true reason=networkTypeChanged 
apn=web.vodafone.de interfaceName=pdp0 networkType=3
07-11 11:21:41.335 I/StatusBarPolicy( 2798): onDataConnectionStateChanged-S:2 
T:3
07-11 11:21:41.335 I/StatusBarPolicy( 2798): updateDataNetType-T:3
07-11 11:21:41.340 D/MobileDataStateTracker( 2689): replacing old 
mInterfaceName (pdp0) with pdp0 for hipri
07-11 11:21:41.360 I/DBG_WSS_TP(14571): [wssHttpAdapter.java Line:2193] 
getIsConnected connect status is false
07-11 11:21:41.375 D/MobileDataStateTracker( 2689): replacing old 
mInterfaceName (pdp0) with pdp0 for supl
07-11 11:21:41.395 D/MobileDataStateTracker( 2689): default Received state= 
CONNECTED, old= CONNECTED, reason= networkTypeChanged, apnTypeList= default,supl
07-11 11:21:41.949 I/WindowOrientationListener( 2689): orientation 17.879604 
gives new rotation = 0
07-11 11:21:41.949 I/OrientationDebug( 2689): [pwm] in 
MyOrientationListener.onOrientationChanged() rotation=0  
mFancyRotationAnimation=0   now call mWindowManager.setRotation()
07-11 11:21:41.949 I/OrientationDebug( 2689): [wms] in setRotation() rotation=0 
alwaysSendConfiguration=false animFlags=0
07-11 11:21:41.949 I/OrientationDebug( 2689): [wms] in setRotation() now call 
setRotationUnchecked()
07-11 11:21:41.954 I/WindowManager( 2689): Setting rotation to 0, animFlags=0
07-11 11:21:41.960 I/TvOut-Observer( 2689): setTvoutOrientation rotation =  0
07-11 11:21:41.960 E/TvOut-Observer( 2689): TvoutSetOrientation
07-11 11:21:41.960 I/ActivityManager( 2689): Config changed: { scale=1.0 
imsi=262/2 loc=de_DE touch=3 keys=1/1/2 nav=1/1 orien=1 layout=34 uiMode=17 
seq=19 FlipFont=0}
07-11 11:21:42.000 I/WebGLAnimator(11619): abortAnimation NONE
07-11 11:21:42.000 W/BitmapWebView(11619): mSystemStatusBarHeight = 38
07-11 11:21:42.000 W/BitmapWebView(11619): set3DSurfaceLayout: displayW/H = 480 
800 parent = 800 202
07-11 11:21:42.075 D/webview (11619): onConfigurationChanged
07-11 11:21:42.080 V/BitmapWebViewLayout(11619): onSizeChanged: old = 800 202, 
new = 480 522
07-11 11:21:42.080 W/BitmapWebView(11619): mSystemStatusBarHeight = 38
07-11 11:21:42.085 W/BitmapWebView(11619): set3DSurfaceLayout: displayW/H = 480 
800 parent = 480 522
07-11 11:21:42.085 V/webview (11619): OnSizeChanged: Enter 
07-11 11:21:42.100 D/FastDormancy( 2811):  before ======= ENTER DORMANCY =======
07-11 11:21:42.100 D/FastDormancy( 2811): [FD] ON default: true
07-11 11:21:42.100 D/FastDormancy( 2811): Before mDormFlag: false in 
getCapaDormancy()
07-11 11:21:42.105 D/FastDormancy( 2811): After mDormFlag: false in 
getCapaDormancy()
07-11 11:21:42.140 D/dalvikvm( 2807): GC_EXTERNAL_ALLOC freed 209K, 55% free 
2828K/6151K, external 6022K/6355K, paused 77ms
07-11 11:21:42.210 E/AutoRotationSettingButton( 2798): de
07-11 11:21:42.220 W/BitmapWebView(11619): mSystemStatusBarHeight = 38
07-11 11:21:42.220 W/BitmapWebView(11619): set3DSurfaceLayout: displayW/H = 480 
800 parent = 480 522
07-11 11:21:42.235 D/Z7Service(14676): Config changed, old locale: de_DE new 
locale:de_DE
07-11 11:21:42.275 D/browser (11619): BrowserActivity IME broadcast receiver = 
true
07-11 11:21:42.275 I/BitmapWebView(11619): pause start
07-11 11:21:42.275 I/BitmapWebView(11619): pause end
07-11 11:21:42.295 V/BitmapWebViewLayout(11619): onSizeChanged: old = 480 522, 
new = 480 442
07-11 11:21:42.295 W/BitmapWebView(11619): mSystemStatusBarHeight = 38
07-11 11:21:42.295 W/BitmapWebView(11619): set3DSurfaceLayout: displayW/H = 480 
800 parent = 480 442
07-11 11:21:42.295 V/webview (11619): OnSizeChanged: Enter 
07-11 11:21:42.295 E/webtextview(11619): ....Doing a 
super.requestRectangleOnScreen....
07-11 11:21:42.295 W/BitmapWebView(11619): mSystemStatusBarHeight = 38
07-11 11:21:42.305 W/BitmapWebView(11619): set3DSurfaceLayout: displayW/H = 480 
800 parent = 480 442
07-11 11:21:42.350 D/dalvikvm( 2807): GC_EXTERNAL_ALLOC freed 58K, 55% free 
2809K/6151K, external 5709K/6011K, paused 61ms
07-11 11:21:42.500 W/BitmapWebView(11619): mSystemStatusBarHeight = 38
07-11 11:21:42.500 W/BitmapWebView(11619): set3DSurfaceLayout: displayW/H = 480 
800 parent = 480 442
07-11 11:21:43.104 D/FastDormancy( 2811):  before ======= ENTER DORMANCY =======
07-11 11:21:43.104 D/FastDormancy( 2811): [FD] ON default: true
07-11 11:21:43.104 D/FastDormancy( 2811): Before mDormFlag: false in 
getCapaDormancy()
07-11 11:21:43.109 D/FastDormancy( 2811): After mDormFlag: false in 
getCapaDormancy()
07-11 11:21:43.445 I/InputReader( 2689): dispatchTouch::touch event's action is 
0
07-11 11:21:43.445 I/InputDispatcher( 2689): Delivering touch to current input 
target: action: 0, channel '4056b310 
com.android.browser/com.android.browser.BrowserActivity (server)'
07-11 11:21:43.445 I/InputDispatcher( 2689): Delivering touch to current input 
target: action: 0, channel 'PointerDevice-11619 (server)'
07-11 11:21:44.110 D/FastDormancy( 2811):  before ======= ENTER DORMANCY =======
07-11 11:21:44.110 D/FastDormancy( 2811): [FD] ON default: true
07-11 11:21:44.110 D/FastDormancy( 2811): Before mDormFlag: false in 
getCapaDormancy()
07-11 11:21:44.110 D/FastDormancy( 2811): After mDormFlag: false in 
getCapaDormancy()
07-11 11:21:44.380 I/InputReader( 2689): dispatchTouch::touch event's action is 
1
07-11 11:21:44.380 I/InputDispatcher( 2689): Delivering touch to current input 
target: action: 1, channel '4056b310 com.android.browser/com.android.browse
```

Original comment by `m.debeni...@gmail.com` on 11 Jul 2011 at 9:32

**Comment by GoogleCodeExporter** [comment]
```
Hi, thanks for the log. Unfortunately, as you said, there's not much useful 
info in it. You seem to have one of those chatty devices that log a lot of 
stuff. That said, the logcat buffer should be a lot bigger than this. You could 
try saving it to a file instead of using email to avoid truncation. If you have 
Dropbox, Google Docs or some editor like AK Notepad, those should show up in 
Log Collector. Try it out to see if you can get some more data. 

You should get something like this from WWWJDIC:

D/BackdoorSearchTask(26681): WWWJDIC URL: 
http://wwwjdic.mygengo.com/cgi-data/wwwjdic?1ZUEcat
D/SearchTask(26681): HTTP timeout: 15000
D/BackdoorSearchTask(26681): WWWJDIC URL: 
http://wwwjdic.mygengo.com/cgi-data/wwwjdic?1ZUEcat

If that doesn't work, I'll send you a version that collects logs without using 
logcat.

Thanks.


```

Original comment by `nikolay....@gmail.com` on 11 Jul 2011 at 11:19

**Comment by GoogleCodeExporter** [comment]
```
I was given a hint that your mobile provider might be caching responses, that's 
why you are getting empty results. I've uploaded an experimental versions that 
tries to get around this, please try it out:

http://code.google.com/p/wwwjdic/downloads/detail?name=Wwwjdic-1.8.4-nocache-sig
ned.apk
```

Original comment by `nikolay....@gmail.com` on 12 Jul 2011 at 5:07

**Comment by GoogleCodeExporter** [comment]
```
I also had this issue but it seems like the experimental version is working. :)
```

Original comment by `ShinNoNoir` on 12 Jul 2011 at 5:22

**Comment by GoogleCodeExporter** [comment]
```
The new version solves the problem. Thank you! I had already saved a new log 
and was about to send it, although again there wasn't much of use in it. This 
time I used aLogCat which can filter  using regular expressions and export to 
file. If you are still interested I can send it.

There is a small quirk in the new version. Today's kanji displays in a smaller 
font and doesn't fill the widget. It's this somehow intended?

```

Original comment by `m.debeni...@gmail.com` on 12 Jul 2011 at 6:38

**Comment by GoogleCodeExporter** [comment]
```
I just removed the widget and put it in place again and it displayed "error". 
Maybe you forgot to set those no cache headers in that code too? Regards.
```

Original comment by `m.debeni...@gmail.com` on 12 Jul 2011 at 6:43

**Comment by GoogleCodeExporter** [comment]
```
Yes, I only set the headers in one place to test if this solves the problem. 
I've uploaded a new binary that (hopefully) sets the proper headers in all the 
proper places. This is not an 'official' release, and I haven't tested it, so 
it might have other problems, but feel free to use it until I release the next 
version.

```

Original comment by `nikolay....@gmail.com` on 12 Jul 2011 at 7:52

**Comment by GoogleCodeExporter** [comment]
```
Thanks. Everything is working perfectly now. Thank you for the quick fixes!
```

Original comment by `m.debeni...@gmail.com` on 12 Jul 2011 at 8:46

**Comment by GoogleCodeExporter** [comment]
```
Dear Nikolay, sorry for my late reply. I was on holiday. I have downloaded this 
version as well and it seems to be working just fine! Great! Thanks for all 
your efforts. Ronald.
```

Original comment by `ronald.h...@j-link.nl` on 12 Jul 2011 at 8:59

**Comment by GoogleCodeExporter** [comment]
```
Fixes are now live in 1.8.5. Should also be a bit faster on some mirrors.

```

Original comment by `nikolay....@gmail.com` on 22 Jul 2011 at 12:58
- Changed state: **Fixed**
